bip78/src/uri.rs

```diff
@@ -88,28 +89,28 @@ impl std::str::FromStr for Uri<'static> {
     }
 }
 
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq)]
 pub enum ParseUriError {
     PjNotPresent,
     Bip21(Bip21Error),
     PayJoin(PjParseError),
 }
 
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq)]
 pub struct Bip21Error(InternalBip21Error);
 
-#[derive(Debug)]
+#[derive(Debug, Eq, PartialEq)]
 pub struct PjParseError(InternalPjParseError);
```

Comment:
I'm not confident in publicly promising equality now. I know it should be no-brainer but sadly `io::Error` exists and it's too popular and it doesn't impl `PartialEq`. Same goes for `std::error::Error`. Promising `PartialEq` means promising the inside will never contain `io::Error`. :(

I want to avoid `io::Error` like a plague but one never knows...

Note that those derives on internal types are OK, it's just public types I'm worried about.
Also feel free to hack together some macro for matching them if it helps in tests.

Reply:
What do you think to add https://docs.rs/assert_matches/1.5.0/assert_matches/ as a dev dep?

Reply:
Sounds great! Trivial to review and being dev-only is a great combination.

Reply:
removed `Eq, PartialEq`, added `assert_matches`

Note I could have matched to the Err variant, but it was a little too verbose because you have to specify the types, so I match against `unwrap_err`.